### PR TITLE
fix: handle invalid kubeconfig file

### LINKED ISSUE #3029: Erroneous kubeconfig breaks kube-context extension
### Bug description

Kubecontext extension is broken when kubeconfig in invalid. Starting it wont help, and then there will be no option to stop or start after that. Also the error is not really clear at first glance. In dev console there is just info level error.

Screenshot_20230627_145158

Log: 
[kube-context-broken.log](https://github.com/containers/podman-desktop/files/11882302/kube-context-broken.log)

### Operating system

Fedora 37, Windows 10

### Installation Method

Other

### Version

next (development version)

### Steps to reproduce

Replace your kube config with this context: 
```
apiVersion: v1
kind: Config
preferences: {}
```

Run podman desktop and take a look at extensions, especially kube-context

### Relevant log output

_No response_

### Additional context

I am not sure how I have achieved this state of kubeconfig. I was cleaning up the older kind configuration and clusters and reinstalling the podman desktop. It happened somehow.

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #3029

### What does this PR do?

Handle errors happening when kubeconfig has invalid content

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

Fixes #3029

### How to test this PR?

Modify content of $HOME/.kube/config and launch Podman Desktop

### CODE REVIEW

File: extensions/kube-context/src/extension.ts [hunk 1]
```diff
@@ -48,45 +48,42 @@ export async function updateContext(
...
-    console.error(`Kubeconfig file at '${kubeconfigFile}' is empty. Skipping.`);
-    return;
-  }
-
   // parse the content using jsYaml
   const kubeConfig = jsYaml.load(kubeConfigRawContent);
 
   // get the current context
-  const currentContext = kubeConfig['current-context'];
+  const currentContext = kubeConfig ? kubeConfig['current-context'] : undefined;
```
Comment (benoitf) [inline extensions/kube-context/src/extension.ts]:
```suggestion
  const currentContext = kubeConfig?.['current-context'];
```

File: extensions/kube-context/src/extension.ts [hunk 2]
```diff
@@ -48,45 +48,42 @@ export async function updateContext(
...
 
   // get all contexts
-  const contexts: KubeContext[] = kubeConfig['contexts'];
+  const contexts: KubeContext[] = kubeConfig?.['contexts'] ? kubeConfig['contexts'] : [];
+
+  if (contexts.length > 0) {
```
Comment (benoitf) [inline extensions/kube-context/src/extension.ts:60]: do we need that check as iterating on an empty array won't do anything

**Review by benoitf — APPROVED** [review]

File: extensions/kube-context/src/extension.ts (hunk 2 above)
Comment (jeffmaury) [reply extensions/kube-context/src/extension.ts:60]: The clue here is that if there are no context then no tray item menu will be registered (as previously when kubeconfig file was empty) thus the test

Comment (benoitf) [reply extensions/kube-context/src/extension.ts:60]: so probably do 

```ts
console.log(`Kubeconfig file at '${kubeconfigFile}' has no contexts. Skipping.`);
if (contexts.length === 0) {
  return;
}
```

**Review by lstocchi — APPROVED** [review]
LGTM

**Pull request merged**